src/component/RoadmapParser.js

```diff
@@ -1,57 +1,113 @@
 import {Roadmap} from "./Roadmap";
+import {RoadmapTask} from "./RoadmapTask";
+import {RoadmapStory} from "./RoadmapStory";
+import {RoadmapAssignee} from "./RoadmapAssignee";
 
 export class RoadmapParser {
-  parse(tasks, people) {
-    let roadmap = new Roadmap();
+  roadmap;
 
-    for (let i = 0; i < tasks.length; i++) {
-      let task = {};
-      task.type = "task";
-      task.group = tasks[i].taskGroup ? tasks[i].taskGroup : '';
-      task.name = tasks[i].name ? tasks[i].name : '';
-      task.style = tasks[i].style ? tasks[i].style : "normal";
-      task.color = tasks[i].color ? tasks[i].color : null;
-      task.order = tasks[i].order ? parseInt(tasks[i].order, 10) : 0;
-      task.from = new Date(tasks[i].from);
-      task.to = new Date(new Date(tasks[i].to).setHours((new Date(tasks[i].to).getHours() + 24)));
-      if (tasks[i].people) {
-        if (Array.isArray(tasks[i].people)) {
-          for (let j = 0; j < tasks[i].people.length; j++) {
-            roadmap.setPerson(
-              this._parsePerson(this._getPersonById(tasks[i].people[j], people), task)
-            );
-          }
-        } else {
-          roadmap.setPerson(
-            this._parsePerson(this._getPersonById(tasks[i].people, people), task)
-          );
-        }
+  /**
+   * @param {object} dataSet
+   * @returns {Roadmap}
+   */
+  parse(dataSet) {
+    this.roadmap = new Roadmap();
+
+    if (dataSet.tasks === undefined) {
+      // invalid
+      return;
+    }
+
+    if (dataSet.stories === undefined && dataSet.assignees === undefined) {
+      // invalid
+      return;
+    }
+
+    if (dataSet.stories) {
+      this._parseStories(dataSet.stories);
+    }
+    if (dataSet.assignees) {
+      this._parseAssignees(dataSet.assignees);
+    }
+    this._parseTasks(dataSet.tasks);
+
+    return this.roadmap;
+  }
+
+  /**
+   * @param {object} stories
+   * @private
+   */
+  _parseStories(stories) {
+    for (let i = 0; i < stories.length; i++) {
+      let story = stories[i];
+      if (story.id === undefined || story.name === undefined) {
+        // invalid item
+        continue;
       }
-      roadmap.setTask(task);
+      this.roadmap.setStory(new RoadmapStory(
```

Comment:
should change this.

1. `setStory` -> `addStory`
2. `setStory(storyId, new Story(...));`